the-blue-alliance-ios/Protocols/Subscribable.swift

```diff
@@ -0,0 +1,24 @@
+import Foundation
+import UIKit
+
+// Subscribable is a protocol view controllers can conform to if they want to have
+// the UI to allow users to subscribe to notifications for models
+// Ex: The 'Event' view controller conforms to Subscribable, which shows the subscribe UI
+
+protocol Subscribable {
+    var favoriteBarButtonItem: UIBarButtonItem! { get }
+}
+
+extension Subscribable where Self: UIViewController, Self: Persistable {
+
+    func presentMyTBAPreferences(modelKey: String, modelType: MyTBAModelType) {
+        let myTBAPreferencesViewController = MyTBAPreferenceViewController(modelKey: modelKey,
```

Comment:
Can we pass a `MyTBASubscribable` generic here and get all the information we need off of it?

Reply:
And if this is true, we can probably remove the parameters for this call and add the property to the protocol